src/microbe_stage/SpawnSystem.cs

```diff
@@ -110,14 +110,14 @@ public void Clear()
     public void DespawnAll()
     {
         ClearSpawnQueue();
-        int despawned = 0;
+        float despawned = 0.0f;
 
         foreach (var spawned in worldRoot.GetChildrenToProcess<ISpawned>(Constants.SPAWNED_GROUP))
         {
             if (!spawned.EntityNode.IsQueuedForDeletion())
             {
                 spawned.DestroyDetachAndQueueFree();
-                ++despawned;
+                despawned += spawned.EntityWeight;
```

Comment:
This line should be before the Destroy call as that should be allowed to mess up the entity entirely.